spec/MessageHandler/Contact/ContactListsSubscriberHandlerSpec.php

```diff
@@ -0,0 +1,138 @@
+        $contactListMapper->mapFromListContactStatusAndSourceId(18, 134, ListSubscriptionStatusResolverInterface::UNSUBSCRIBED_STATUS_CODE)->willReturn($fourthContactList);
+
+        $activeCampaignContactListClient->create($firstContactList)->willReturn($createResourceResponse);
+        $activeCampaignContactListClient->create($fourthContactList)->willThrow(new HttpException(200));
+
+        $this->beConstructedWith($customerRepository, $customerChannelsResolver, $listSubscriptionStatusResolver, $activeCampaignContactListClient, $contactListMapper, $logger);
+    }
+
+    public function it_is_initializable(): void
+    {
+        $this->shouldBeAnInstanceOf(ContactListsSubscriberHandler::class);
+    }
+
+    public function it_throws_if_customer_is_not_found(
+        CustomerRepositoryInterface $customerRepository
+    ): void {
+        $customerRepository->find(12)->shouldBeCalledOnce()->willReturn(null);
+
+        $this->shouldThrow(new InvalidArgumentException('Customer with id "12" does not exists.'))->during(
+            '__invoke',
+            [new ContactListsSubscriber(12)]
+        );
+    }
+
+    public function it_throws_if_customer_is_not_an_implementation_of_customer_active_campaign_aware_interface(
+        CustomerRepositoryInterface $customerRepository,
+        SyliusCustomerInterface $syliusCustomer
+    ): void {
+        $customerRepository->find(12)->shouldBeCalledOnce()->willReturn($syliusCustomer);
+
+        $this->shouldThrow(new InvalidArgumentException('The Customer entity should implement the "Webgriffe\SyliusActiveCampaignPlugin\Model\CustomerActiveCampaignAwareInterface" class.'))->during(
+            '__invoke',
+            [new ContactListsSubscriber(12)]
+        );
+    }
+
+    public function it_throws_if_customer_has_not_been_exported_to_active_campaign_yet(
+        ActiveCampaignAwareInterface $customer
+    ): void {
+        $customer->getActiveCampaignId()->willReturn(null);
+
+        $this->shouldThrow(new InvalidArgumentException('The Customer with id "12" does not have an ActiveCampaign id.'))->during(
+            '__invoke',
+            [new ContactListsSubscriber(12)]
+        );
+    }
+
+    public function it_throws_if_http_exception_does_not_have_200_as_status_code(
+        ActiveCampaignResourceClientInterface $activeCampaignContactListClient,
+        ContactListInterface $fourthContactList
+    ): void {
+        $activeCampaignContactListClient->create($fourthContactList)->willThrow(new HttpException(400));
+
+        $this->shouldThrow(new HttpException(400))->during(
+            '__invoke',
+            [new ContactListsSubscriber(12)]
+        );
+    }
+
+    public function it_handles_customer_list_subscriptions(
+        LoggerInterface $logger
+    ): void {
+        $logger->info('The association with the list with id "18" already exists for the contact with id "134".')->shouldBeCalledOnce();
```

Comment:
We should assert that
```
$activeCampaignContactListClient->create($firstContactList)->willReturn($createResourceResponse);
```
is called.
An assertion only on the logger is not really solid.

Reply:
Done https://github.com/webgriffe/SyliusActiveCampaignPlugin/commit/0fe91899f5429943b77e2b9d0c073335d60b7813.